packages/chain-controller/src/SnapChainProviderClient.test.ts

```diff
@@ -0,0 +1,74 @@
+import { ChainRpcMethod } from '@metamask/chain-api';
+import type { SnapId } from '@metamask/snaps-sdk';
+import { HandlerType } from '@metamask/snaps-utils';
+import type { Json } from '@metamask/utils';
+
+import { SnapChainProviderClient } from './SnapChainProviderClient';
+import { SnapHandlerClient } from './SnapHandlerClient';
+
+describe('SnapChainProviderClient', () => {
+  const handleRequest = jest.fn();
+
+  const snapId = 'local:localhost:3000' as SnapId;
+  const snapController = {
+    handleRequest,
+  };
+  const snapClient = new SnapHandlerClient({
+    handler: handleRequest,
+    snapId,
+  });
+
+  const makeRequest = ({
+    method,
+    params,
+  }: {
+    method: string;
+    params: Json;
+  }) => {
+    return {
+      snapId,
+      origin: 'metamask',
+      handler: HandlerType.OnRpcRequest,
+      request: {
+        id: expect.any(String),
+        jsonrpc: '2.0',
+        method,
+        params,
+      },
+    };
+  };
+
+  describe('getBalances', () => {
+    it('dispatch chain_getBalances', async () => {
+      const client = new SnapChainProviderClient(snapClient);
+
+      const address = 'bc1qrp0yzgkf8rawkuvdlhnjfj2fnjwm0m8727kgah';
+      const scope = 'bip122:000000000019d6689c085ae165831e93';
+      const asset = `${scope}/asset:0`;
+
+      const request = makeRequest({
+        method: ChainRpcMethod.GetBalances,
+        params: {
+          scope,
+          accounts: [address],
+          assets: [asset],
+        },
+      });
+
+      const response = {
+        balances: {
+          [address]: {
+            [asset]: {
+              amount: '70.02255139',
+            },
+          },
+        },
+      };
+
+      snapController.handleRequest.mockResolvedValue(response);
+      const result = await client.getBalances(scope, [address], [asset]);
+      expect(snapController.handleRequest).toHaveBeenCalledWith(request);
+      expect(result).toStrictEqual(response);
```

Comment:
Nit: To help make it abundantly clear what exactly is being tested here, what do you think about using line breaks to divide the test into different "phases"? For instance, in this case you could remove the empty lines between all of the steps above this line, and then surround this line with empty lines to separate it visually. See here for more details: https://github.com/MetaMask/contributor-docs/blob/main/docs/unit-testing.md#-highlight-the-exercise-phase

``` suggestion
      const client = new SnapChainProviderClient(snapClient);
      const address = 'bc1qrp0yzgkf8rawkuvdlhnjfj2fnjwm0m8727kgah';
      const scope = 'bip122:000000000019d6689c085ae165831e93';
      const asset = `${scope}/asset:0`;
      const request = makeRequest({
        method: ChainRpcMethod.GetBalances,
        params: {
          scope,
          accounts: [address],
          assets: [asset],
        },
      });
      const response = {
        balances: {
          [address]: {
            [asset]: {
              amount: '70.02255139',
            },
          },
        },
      };
      snapController.handleRequest.mockResolvedValue(response);

      const result = await client.getBalances(scope, [address], [asset]);

      expect(snapController.handleRequest).toHaveBeenCalledWith(request);
      expect(result).toStrictEqual(response);
```

Reply:
If this match our conventions, then yes why not!